src/Storages/Kafka/ReadBufferFromKafkaConsumer.cpp

```diff
@@ -467,8 +468,26 @@ bool ReadBufferFromKafkaConsumer::nextImpl()
         return false;
 
     // XXX: very fishy place with const casting.
-    auto * new_position = reinterpret_cast<char *>(const_cast<unsigned char *>(current->get_payload().get_data()));
-    BufferBase::set(new_position, current->get_payload().get_size(), 0);
+    auto message_data = current->get_payload().get_data();
+
+    char * new_position;
+    size_t message_size;
+
+    if (unlikely(message_data == NULL))
+    {
+        // in some cases message can be NULL (tombstone records for example)
+        // parsers are not ready to get NULLs on input, so the best we can do here
+        // it substitute NULL with an pointer to empty dummy buffer.
+        new_position = dummy_buf.get();
```

Comment:
It should be Ok to have empty buffer pointing to nullptr.

Reply:
Sounds reasonable, but it is not always true (it happened not the first time, in older versions, other formats were also broken this way, see  #19255). Unless we have some automatic (unit?) tests checking that all the formats in the factory satisfy that invariant - we can't be sure. AFAIK that never happens outside of Kafka's use case, so it's easier to follow the behavior of other buffers.

Reply:
Also that makes the fix very local and simple to backport.